crypto/rsa/rsa_chk.c

```diff
@@ -30,10 +30,13 @@ int RSA_check_key_ex(const RSA *key, BN_GENCB *cb)
     }
 
     /* multi-prime? */
-    if (key->version == RSA_ASN1_VERSION_MULTI
-        && (ex_primes = sk_RSA_PRIME_INFO_num(key->prime_infos)) <= 0) {
-        RSAerr(RSA_F_RSA_CHECK_KEY_EX, RSA_R_INVALID_MULTI_PRIME_KEY);
-        return 0;
+    if (key->version == RSA_ASN1_VERSION_MULTI) {
+        ex_primes = sk_RSA_PRIME_INFO_num(key->prime_infos);
+        if (ex_primes <= 0
```

Comment:
should this be a macro?  it's used a couple of times.

Reply:
Well, wouldn't such macro fall into undesired "affect-control-flow" category?

[On side note I'd argue that `prime_infos` doesn't have to and shouldn't be a stack. I mean stack effectively adds essentially unnecessary level of indirection and memory fragmentation...]

Reply:
Sorry, I meant the if() test itself. And I totally agree with your side-note.